inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -1142,7 +1142,11 @@ private <T> void destroyBean(@NonNull BeanRegistration<T> registration, boolean
         beanToDestroy = triggerPreDestroyListeners(definition, beanToDestroy);
 
         if (definition instanceof DisposableBeanDefinition) {
-            ((DisposableBeanDefinition<T>) definition).dispose(this, beanToDestroy);
+            try {
+                ((DisposableBeanDefinition<T>) definition).dispose(this, beanToDestroy);
+            } catch (Exception e) {
+                LOG.info("Error disposing bean [{}]... Continuing...", beanToDestroy, e);
```

Comment:
Should be `warn` and I don't think you can use parameterization like this to include the stack trace. Has to be normal string concatenation. Also include `e.getMessage()` in the error

Reply:
I think it will work, the last parameter can be the exception cause

Reply:
@dstepanov is right 😄 

Changed it to:

```
                if (LOG.isWarnEnabled()) {
                    LOG.warn("Error disposing bean [" + beanToDestroy.toString() + "]... Continuing...", e);
                }
```